core/src/main/scala/zio/config/PropertyTree.scala

```diff
@@ -292,10 +296,13 @@ object PropertyTree {
   def unflatten[K, V](map: Map[Vector[K], ::[V]]): List[PropertyTree[K, V]] =
     mergeAll(map.toList.map(tuple => unflatten(tuple._1.toList, tuple._2)))
 
-  def mergeAll[K, V](list: List[PropertyTree[K, V]]): List[PropertyTree[K, V]] =
-    list.foldRight(List[PropertyTree[K, V]](PropertyTree.empty)) {
-      case (tree, acc) => acc.flatMap(tree0 => tree.merge(tree0))
-    }
+  def mergeAll[K, V](list: List[PropertyTree[K, V]]): List[PropertyTree[K, V]] = list.reverse match {
+    case Nil => PropertyTree.empty :: Nil
```

Comment:
If we need a guarantee that `mergeAll` produces non empty, we should return `::` here. But I'm not sure we need that guarantee since a user can always do that themselves with the return value of `mergeAll`. So you could return `Nil` here and fix anything if it breaks.

Reply:
done